experimenter/evaluate_pipeline_new.py

```diff
@@ -0,0 +1,199 @@
+import itertools as it
+import json
+import yaml
+import os
+import parser
+
+from typing import Any, List, Tuple
+from uuid import UUID
+from experimenter import config, utils
+
+from d3m import cli as d3m_cli
+from d3m.contrib.pipelines import K_FOLD_TABULAR_SPLIT_PIPELINE_PATH as k_fold_split_path
+from d3m.contrib.pipelines import FIXED_SPLIT_TABULAR_SPLIT_PIPELINE_PATH as fixed_split_path
+from experimenter.databases.d3m_mtl import D3MMtLDB
+
+
+def save_pipeline_run_to_d3m_db(pipeline_run_path: str):
+    """ 
+    Saves a pipeline run document to the d3m database.
+
+    Parameters
+    ----------
+    pipeline_run_path : path_like str
+        path to pipeline_run document
+
+    Returns:
+    ----------
+    TODO
+
+    Raises:
+    ----------
+    TODO
+    """
+    d3m_db = D3MMtLDB()
+    with open(pipeline_run_path) as pipeline_data:
+        pipeline_run = yaml.full_load(pipeline_data)
+    return D3MMtLDB().save_pipeline_run(pipeline_run)
+
+def evaluate_pipeline_on_problem(pipeline: str=None,
+    problem: str=None,
+    input: str=None,
+    random_seed: int=0,
+    data_pipeline_path: str=k_fold_split_path,
+    data_random_seed: int=0,
+    data_params=None,
+    scoring_pipeline: str=None,
+    scoring_params=None,
+    scoring_random_seed: int=0):
+    """ 
+    Evaluate pipeline on problem.
+    A less verbose form of running d3m's runtime cli 'evaluate' command.
+    See 'evaluate_pipeline_via_d3m_cli' for more options for running 
+    the 'evaluate' command.
+
+    Parameters
+    ----------
+    pipeline_path : path_like str
+        path to pipeline doc
+    problem_path : path_like str 
+        path to problem doc
+    input_path : path_like str
+        path to input full data
+    random_seed : int   
+        random seed to be used for pipeline run
+    data_pipeline_path: str
+        path to data prepation pipeline
+    data_random_seed: int
+        random_seed to be used in data preparation
+    data_params:
+        parameters for data preparation
+    scoring_params:
+        parameters for scoring pipeline
+    scoring_random_seed: int
+        random seed for scoring
+    scoring_pipeline: str
+        path to scoring pipeline
+
+    Returns:
+    ----------
+    None
+
+    Raises:
+    ---------------------------------
+    OSError
+        when a file cannot be opened
+    """
+    output_run_path = []
+    with open(pipeline, 'r') as data:
+        pipe = json.load(data)
+        output_run_path.append(pipe['id'])
+    with open(problem, 'r') as data:
+        prob = json.load(data)
+        output_run_path.append(prob['about']['problemID'])
+    output_run_path.append(str(random_seed))
+    #get the output run path
+    output_run_path = os.path.abspath(os.path.join('/data', 'Pipeline_Run', 
+                                                   '_'.join(output_run_path)+'.yaml'))
```

Comment:
Creating the pipeline run path should be a separate function. It may be useful elsewhere.